### PR TITLE
feat(wallet): Wallet V2 Line Chart

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Updates the Wallet LineChart to the new Wallet V2 designs

Resolves <https://github.com/brave/brave-browser/issues/29834>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to the `Portfolio` Page
2. The LineChart should be updated to the new design
3. Select an `Asset`
4. The LineChart should be updated to the new design

https://user-images.githubusercontent.com/40611140/232895256-47896114-264c-45e3-8c6c-30ddc0adf2c6.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/line-chart/custom-tooltip/index.tsx [hunk 1]
```diff
@@ -5,53 +5,80 @@
...
+  // Selectors
+  const defaultFiatCurrency =
+    useSafeWalletSelector(WalletSelectors.defaultFiatCurrency)
+
+  // Methods
   const parseDate = (date: Date) => {
     const formatedDate = new Date(date).toLocaleDateString('en-US', { month: 'numeric', day: 'numeric', year: 'numeric' })
     const formatedTime = new Date(date).toLocaleTimeString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true })
     return `${formatedDate} ${formatedTime}`
   }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/line-chart/custom-tooltip/index.tsx:43]: pref(non-blocking): could we move this function declaration out of the component?

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/line-chart/custom-tooltip/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/line-chart/custom-tooltip/index.tsx]: Moved to Utils with a test!

File: components/brave_wallet_ui/components/desktop/line-chart/custom-reference-dot.tsx [hunk 2]
```diff
@@ -3,66 +3,40 @@
...
+  onUpdateXPosition,
+}: Props) => {
+
+  // Effects
+  React.useLayoutEffect(() => {
+    if (cy !== undefined && cx !== undefined) {
+      onUpdateYPosition(cy + 10)
+      onUpdateXPosition(cx)
+    }
+  }, [cy, cx, onUpdateYPosition])
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/line-chart/custom-reference-dot.tsx]: Updated!

File: components/brave_wallet_ui/utils/datetime-utils.ts [hunk 3]
```diff
@@ -48,6 +48,32 @@ export function formatDateAsRelative (date: Date, now: Date = new Date()) {
...
  * @return {!Date}
  */
- export function serializedTimeDeltaToJSDate (timeDelta: SerializableTimeDelta): Date {
+export function serializedTimeDeltaToJSDate (
+  timeDelta: SerializableTimeDelta
+): Date {
   return new Date(Number(timeDelta.microseconds) / 1000)
 }
+
+export const formateTimelineDate = (date: Date) => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/utils/datetime-utils.ts:57]: typo
```suggestion
export const formatTimelineDate = (date: Date) => {
```

**Review by josheleonard — CHANGES_REQUESTED** [review]
typo in function name

File: components/brave_wallet_ui/utils/datetime-utils.ts [hunk 4]
```diff
@@ -48,6 +48,32 @@ export function formatDateAsRelative (date: Date, now: Date = new Date()) {
...
+  timeDelta: SerializableTimeDelta
+): Date {
   return new Date(Number(timeDelta.microseconds) / 1000)
 }
+
+export const formatTimelineDate = (date: Date) => {
+  const formatedDate =
+    new Date(date)
+      .toLocaleDateString(
+        'en-US',
```
Comment (onyb) [inline components/brave_wallet_ui/utils/datetime-utils.ts:61]: Shouldn't this be based on the user's locale? For example, `mm/dd/yyyy` is used only in the US.

**Pull request merged**